ddsrecorder_participants/src/cpp/recorder/monitoring/producers/DdsRecorderStatusMonitorProducer.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <ddspipe_core/monitoring/consumers/DdsMonitorConsumer.hpp>
+#include <ddspipe_core/monitoring/consumers/StdoutMonitorConsumer.hpp>
+
+#include <ddsrecorder_participants/recorder/monitoring/producers/DdsRecorderStatusMonitorProducer.hpp>
+
+namespace eprosima {
+namespace ddsrecorder {
+namespace participants {
+
+void DdsRecorderStatusMonitorProducer::register_consumer(
+        std::unique_ptr<ddspipe::core::IMonitorConsumer<DdsRecorderMonitoringStatus>> consumer)
+{
+    if (!enabled_)
+    {
+        logWarning(DDSPIPE_MONITOR, "MONITOR | Not registering consumer " << consumer->get_name() << " on "
+                "DdsRecorderStatusMonitorProducer since the DdsRecorderStatusMonitorProducer is disabled.");
+
+        return;
+    }
+
+    logInfo(DDSPIPE_MONITOR, "MONITOR | Registering consumer " << consumer->get_name() << " on DdsRecorderStatusMonitorProducer.");
+
+    consumers_.push_back(std::move(consumer));
+}
+
+void DdsRecorderStatusMonitorProducer::produce()
+{
+    if (!enabled_)
+    {
+        // Don't consume if the producer is not enabled
+        return;
+    }
+
+    // Take the lock to prevent:
+    //      1. Changing the data while it's being saved.
+    //      2. Simultaneous calls to produce.
+    std::lock_guard<std::mutex> lock(mutex_);
+
+    logInfo(DDSPIPE_MONITOR, "MONITOR | Producing DdsRecorderMonitoringStatus.");
+
+    data_.error_status(error_status_);
+    data_.ddsrecorder_error_status(ddsrecorder_error_status_);
+    data_.has_errors(has_errors_);
+}
+
+void DdsRecorderStatusMonitorProducer::consume()
+{
+    if (!enabled_)
+    {
+        // Don't consume if the producer is not enabled
+        return;
+    }
+
+    logInfo(DDSPIPE_MONITOR, "MONITOR | Consuming DdsRecorderMonitoringStatus.");
```

Comment:
Change the category to `DDSRECORDER_MONITOR`? Or maybe we would prefer to collapse all categories into a single `MONITOR`, or `MONITOR_SomethingElse` (easier to filter with regex).

Reply:
You're right, it should be `DDSRECORDER_MONITOR`. To make it easier to filter, all the monitoring logs use the event "MONITOR" (the part of the message to the left of the "|").

Reply:
Some changes still pending.